sklearn/metrics/_plot/tests/test_plot_precision_recall.py

```diff
@@ -0,0 +1,117 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_allclose
+
+from sklearn.base import BaseEstimator
+from sklearn.metrics import plot_precision_recall_curve
+from sklearn.metrics import average_precision_score
+from sklearn.metrics import precision_recall_curve
+from sklearn.datasets import load_breast_cancer
+from sklearn.datasets import load_iris
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.linear_model import LogisticRegression
+from sklearn.exceptions import NotFittedError
+
+
+@pytest.fixture(scope="module")
+def data_binary():
+    return load_breast_cancer(return_X_y=True)
+
+
+def test_error_non_binary(pyplot):
+    X, y = load_iris(return_X_y=True)
+    clf = DecisionTreeClassifier()
+    clf.fit(X, y)
+
+    msg = "multiclass format is not supported"
+    with pytest.raises(ValueError, match=msg):
+        plot_precision_recall_curve(clf, X, y)
+
+    msg = "Estimator should solve a binary classification problem"
+    y_binary = y == 1
```

Comment:
I don't understand why this raises this error, both semantically and why that is what the code does. I thought the code checked ``y_pred``, which we're not changing here, right?